execution/function/histogram.go

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) The Thanos Community Authors.
+// Licensed under the Apache License 2.0.
```

Comment:
I think it is okay to have a separate operator for this function. Just wondering you make it a separate opterator because of the function arguments?
What about other functions that take different args. Shall we use a new operator every time?

Reply:
It's not just the arguments, but also this operator has its own pool, it has separate series as outputs (we need to drop the "le" label) and it has a different Next function. So everything is really different which is why I thought it is cleaner to have a separate operator.